python/semantic_kernel/data/vector_record_store_base.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from abc import ABC, abstractmethod
+from inspect import signature
+from typing import Any, Generic, TypeVar
+
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.data.models.vector_store_model_definition import (
+    VectorStoreContainerDefinition,
+    VectorStoreRecordDefinition,
+)
+from semantic_kernel.data.models.vector_store_record_fields import (
+    VectorStoreRecordDataField,
+    VectorStoreRecordVectorField,
+)
+from semantic_kernel.data.protocols.vector_store_model_serde_protocols import (
+    VectorStoreModelFunctionSerdeProtocol,
+    VectorStoreModelPydanticProtocol,
+    VectorStoreModelToDictFromDictProtocol,
+)
+from semantic_kernel.exceptions.memory_connector_exceptions import (
+    MemoryConnectorException,
+    VectorStoreModelDeserializationException,
+    VectorStoreModelSerializationException,
+)
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel", bound=object)
```

Comment:
Just a question: since everything in Python is an object, what effect does `bound=object` have here?